tests/test_cache.py

```diff
@@ -0,0 +1,46 @@
+import redis
+import unittest
+from cachecontrol.caches.file_cache import FileCache
+from cachecontrol.caches.redis_cache import RedisCache
+from canonicalwebteam.http import CachedSession
+from datetime import timedelta
+from unittest import TestCase
+
+
+def create_redis_connection_pool(
+    redis_port="6379", redis_host="localhost", redis_db=0
+):
+    return redis.ConnectionPool(host=redis_host, port=redis_port, db=redis_db)
+
+
+class TestCachedSession(TestCase):
+    def test_custom_heuristic(self):
+        session = CachedSession(fallback_cache_duration=10)
+        adapter = session.get_adapter("http://")
+        heuristic = adapter.heuristic
+        delta = heuristic.delta
+        self.assertEqual(delta, timedelta(seconds=10))
+
+    def test_default_heuristic(self):
+        session = CachedSession()
+        adapter = session.get_adapter("http://")
+        heuristic = adapter.heuristic
```

Comment:
Rather than deep-inspecting heuristics, it would be much better to actually test the external behaviour of the module.

In my initial experiments with how to do this, I tried to use [responses](https://github.com/getsentry/responses) (which we've used before to mock HTTP responses), and [requests-mock](https://requests-mock.readthedocs.io/en/latest/). Neither of these worked - they seem to override the caching adapter.

But then I tried httpretty, which works a charm. E.g.:

``` python
httpretty.register_uri(
    httpretty.GET,
    "https://httpbin.org/ip",
    body='{"origin": "127.0.0.1"}',
    adding_headers={"Cache-Control": "max-age=30"},
)
```

Could you try using this to test the library from the outside?

Reply:
I like the idea (from httpretty README) of mocking `https://httpbin.org` endpoints, so you can write tests in such a way that they would still work even without the mocking layer. That seems neat to me.

Reply:
Used it, works great. Throws some errors during test execution though. Specifically:
```
ResourceWarning: unclosed file <_io.BufferedRandom name=9>
  self.fd = FakeSockFile()
```

I have opened an issue at https://github.com/gabrielfalcao/HTTPretty/issues/368